docs/content/docs/reference/react/hooks.mdx

```diff
@@ -0,0 +1,393 @@
+
+#### Usage
+
+```jsx
+import { useState } from 'react'
+import { useEmailPasswordSignUp } from '@nhost/react'
+
+const Component = () => {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const { signUp, isLoading, isSuccess, needsVerification, isError, error } =
+    useEmailPasswordSignUp(email, password)
+  return (
+    <div>
+      <input value={email} onChange={(event) => setEmail(event.target.value)} placeholder="Email" />
+      <input
+        value={password}
+        onChange={(event) => setPassword(event.target.value)}
+        placeholder="Password"
+      />
+      <button onClick={signUp}>Register</button>
+      {isSuccess && <div>Your account have beed created! You are now authenticated</div>}
+      {needsVerification && (
+        <div>
+          Please check your mailbox and follow the verification link to confirm your registration
+        </div>
+      )}
+    </div>
+  )
+}
+```
+
+### Email + password authentication
+
+```js
+const { signIn, isLoading, needsVerification, isSuccess, isError, error } =
+  useEmailPasswordSignIn(email?: string, password?: string)
+```
+
+- `signIn: (email?: string, password?: string) => void` will try to authenticate. The email/password arguments will take precedence over the possible state values used when creating the hook.
+- `isLoading: boolean` returns `true` when the action is executing, `false` when the action result is available.
+- `needsVerification: boolean` returns `true` if the user email is still pending verification.
+- `isSuccess: boolean` returns `true` if the user has successfully authenticated. Returns `false` in case or error or if the new email needs to be verified first.
+- `isError: boolean` returns `true` if an error occurred.
+- `error?: {status: number, error: string, message: string}` provides details about the error.
+
+#### Usage
+
+```jsx
+import { useState } from 'react'
+import { useEmailPasswordSignIn } from '@nhost/react'
+
+const Component = () => {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const { signIn, isLoading, isSuccess, needsVerification, isError, error } =
+    useEmailPasswordSignIn(email, password)
+  return (
+    <div>
+      <input value={email} onChange={(event) => setEmail(event.target.value)} placeholder="Email" />
+      <input
+        value={password}
+        onChange={(event) => setPassword(event.target.value)}
+        placeholder="Password"
+      />
+      <button onClick={signUp}>Register</button>
+      {isSuccess && <div>Authentication suceeded</div>}
+      {needsVerification && (
+        <div>
+          Your email is still pending verification. Check your mailbox and follow the instructions
+          to activate your account.
```

Comment:
Suggestion: "You must verify your email to sign in. Check your mailbox and follow the instructions to verify your email".